modules/widgets/internet-defense-league.php

```diff
@@ -0,0 +1,154 @@
+		);
+
+		$this->variants = array(
+			'banner' => __( 'Banner at the top of my site', 'jetpack' ),
+			'modal'  => __( 'Modal (Overlay Box)', 'jetpack' ),
+		);
+
+		$this->badges = array(
+			'shield_badge'   => __( 'Shield Badge', 'jetpack' ),
+			'super_badge'    => __( 'Super Badge', 'jetpack' ),
+			'side_bar_badge' => __( 'Red Cat Badge', 'jetpack' ),
+		);
+
+		if ( $this->no_current === false ) {
+			$this->badges['none'] = __( 'Don\'t display a badge (just the campaign)', 'jetpack' );
+		}
+
+		$this->defaults = array(
+			'campaign' => key( $this->campaigns ),
+			'variant'  => key( $this->variants ),
+			'badge'    => key( $this->badges ),
+		);
+	}
+
+	public function widget( $args, $instance ) {
+		$instance = wp_parse_args( $instance, $this->defaults );
+
+		if ( 'none' != $instance['badge'] ) {
+			if ( ! isset( $this->badges[ $instance['badge'] ] ) ) {
+				$instance['badge'] = $this->defaults['badge'];
+			}
+			$badge_url = esc_url( 'https://internetdefenseleague.org/images/badges/final/' . $instance['badge'] . '.png' );
+			$photon_badge_url = jetpack_photon_url( $badge_url );
+			echo $args['before_widget'];
+			echo '<p><a href="https://internetdefenseleague.org/"><img src="' . $photon_badge_url . '" alt="Member of The Internet Defense League" style="max-width: 100%; height: auto;" /></a></p>';
+			echo $args['after_widget'];
+		}
+
+		if ( 'none' != $instance['campaign'] ) {
+			$this->campaign = $instance['campaign'];
+			$this->variant  = $instance['variant'];
+			add_action( 'wp_footer', array( $this, 'footer_script' ) );
+		}
+
+		/** This action is already documented in modules/widgets/gravatar-profile.php */
+		do_action( 'jetpack_stats_extra', 'widget_view', 'internet_defense_league' );
+	}
+
+	public function footer_script() {
+		if ( ! isset( $this->campaigns[ $this->campaign ] ) ) {
+			$this->campaign = $this->defaults['campaign'];
+		}
+
+		if ( ! isset( $this->variants[ $this->variant ] ) ) {
+			$this->variant = $this->defaults['variant'];
+		}
+		?>
+		<script type="text/javascript">
+			window._idl = {};
+			_idl.campaign = "<?php echo esc_js( $this->campaign ); ?>";
+			_idl.variant = "<?php echo esc_js( $this->variant ); ?>";
+			(function() {
+				var idl = document.createElement('script');
+				idl.type = 'text/javascript';
+				idl.async = true;
+				idl.src = ('https:' == document.location.protocol ? 'https://' : 'http://') + 'members.internetdefenseleague.org/include/?url=' + (_idl.url || '') + '&campaign=' + (_idl.campaign || '') + '&variant=' + (_idl.variant || 'banner');
+				document.getElementsByTagName('body')[0].appendChild(idl);
+			})();
+		</script>
+		<?php
+	}
+
+	public function form( $instance ) {
+		$instance = wp_parse_args( $instance, $this->defaults );
+
+		// Hide first two form fields if no current campaigns.
+		if ( false === $this->no_current ) {
+			echo '<p><label>';
+			echo __( 'Which Internet Defense League campaign do you want to participate in?', 'jetpack' ) . '<br />';
```

Comment:
This and similar instances should use `esc_html_e()` (or `_e()` when necessary) instead of `echo __()`